### DISCUSSION TITLE
COMPILE ERROR

### DISCUSSION THREAD

**Discussion by Trycoz** [IIC2133-PUC/2024-1#165] [Tarea 2] [answered]
Hola!

Viendo los rankings me di cuenta de que tengo un error de COMPILE ERROR, siendo que en mi computador corren los tests bien. Hay algo que tenga que cambiar para evitar esto? (hice make clean && make pero no se que más se puede hacer :( )

**Comment by benjavicente** [comment, marked as answer]
Hola! El ranking preliminar lo estoy obteniendo en MacOS. El compilador de Mac es menos permisivo en compilar cosas que no son estándar de C. En tu caso, el problema está en que defines funciones dentro de funciones, lo que no es propio del estandar de C.

La revisión final no es en MacOS, pero el preliminar sí.

Puedes usar una versión más estricta del compilador en Linux cambiando esto en el Makefile:

```diff
- CC=gcc -std=gnu11
+ CC=gcc -std=gnu11 -pedantic-errors
```
